facade/src/main/java/org/jboss/pnc/facade/deliverables/DeliverableAnalyzerManagerImpl.java

```diff
@@ -673,17 +700,25 @@ private String createGenericPurl(String filename, String sha256) {
         }
     }
 
-    private String createIdentifier(MavenArtifact mavenArtifact) {
-        return Arrays
-                .asList(
-                        mavenArtifact.getGroupId(),
-                        mavenArtifact.getArtifactId(),
-                        mavenArtifact.getType(),
-                        mavenArtifact.getVersion(),
-                        mavenArtifact.getClassifier())
-                .stream()
-                .filter(Objects::nonNull)
-                .collect(Collectors.joining(":"));
+    private String createIdentifier(Artifact artifact) {
+        if (artifact instanceof MavenArtifact) {
+            MavenArtifact mavenArtifact = (MavenArtifact) artifact;
+            return Arrays
+                    .asList(
+                            mavenArtifact.getGroupId(),
+                            mavenArtifact.getArtifactId(),
+                            mavenArtifact.getType(),
+                            mavenArtifact.getVersion(),
+                            mavenArtifact.getClassifier())
+                    .stream()
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.joining(":"));
+        } else if (artifact instanceof WindowsArtifact) {
+            WindowsArtifact windowsArtifact = (WindowsArtifact) artifact;
+            return String.join("-", windowsArtifact.getName(), windowsArtifact.getVersion());
```

Comment:
Not sure.

Reply:
@pkocandr For Windows, I just chose `<name>-<version>` as the identifier. I wonder if it's similar for NPM. It obviously will not match the Maven identifier format.